Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/README.md

```diff
@@ -0,0 +1,298 @@
+
+### Single Route Configuration
+```
+{
+    "LambdaResourceName": "AddLambdaFunction",
+    "HttpMethod": "Get",
+    "Path": "/add/{x}/{y}"
+}
+```
+
+### Multiple Routes Configuration
+
+```
+[
+    {
+        "LambdaResourceName": "AddLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/add/{x}/{y}"
+    },
+    {
+        "LambdaResourceName": "SubtractLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/minus/{x}/{y}"
+    }
+]
+
+```
+
+## Running the Test Tool Project via an IDE
+
+Download this repository and create/modify the launch settings `Properties/launchSettings.json` to startup the test tool
+
+```json
+{
+    "profiles": {
+        "Lambda Test Tool": {
+          "commandName": "Project",
+            "commandLineArgs": "start --lambda-emulator-port 5050 --api-gateway-emulator-port 5051 --api-gateway-emulator-mode Rest",
+            "environmentVariables": {
+                "APIGATEWAY_EMULATOR_ROUTE_CONFIG": {
+                    "LambdaResourceName": "AddLambdaFunction",
+                    "HttpMethod": "Get",
+                    "Path": "/add/{x}/{y}"
+                }
+            }
+        }
+    }
+}
+
+```
+
+## Example Lambda Function Setup
+
+Here's a simple Lambda function that adds two numbers together. This can be implemented in two ways:
+
+### 1. Lambda Function Code
+Here's a simple Lambda function that adds two numbers together:
+
+#### Option 1: Using Top-Level Statements
+
+
+```csharp
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.RuntimeSupport;
+using Amazon.Lambda.Serialization.SystemTextJson;
+
+var Add = (APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) =>
+{
+    // Parse x and y from the path parameters
+    var x = int.Parse(request.PathParameters["x"]);
+    var y = int.Parse(request.PathParameters["y"]);
+    return (x + y).ToString();
+};
+
+await LambdaBootstrapBuilder.Create(Add, new CamelCaseLambdaJsonSerializer())
+    .Build()
+    .RunAsync();
+
+```
+
+#### Option 2: Using Class Library
+```
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+
+namespace MyLambdaFunction;
+
+public class Function
+{
+    public int Add(APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context)
+    {
+        var x = int.Parse(request.PathParameters["x"]);
+        var y = int.Parse(request.PathParameters["y"]);
+        return x + y;
+    }
+}
+```
+
+### 2. Configuration Files
+**Properties/launchSettings.json**
+Configure the Lambda function to use the test tool:
+
+```
+{
+  "profiles": {
+    "AspireTestFunction": {
+      "commandName": "Project",
+      "environmentVariables": {
+        "AWS_LAMBDA_RUNTIME_API": "localhost:5050/AddLambdaFunction"
+      }
+    }
+  }
+}
+```
+
+**aws-lambda-tools-defaults.json**
+
+For top-level statements, your `aws-lambda-tools-defaults.json` should be:
```

Comment:
As of right now our test tool or aspire integration don't do anything with `aws-lambda-tools-defaults.json` so I wouldn't talk about here.

Reply:
Doesnt the function handler name need to match in order for api gateway to work with it?

Reply:
maybe it was just the `AWS_LAMBDA_RUNTIME_API` function name that needs to match. ill retest

Reply:
i removed this part